### PR TITLE
Add support for NXP S32 System Timer Module driver

### PR DESCRIPTION
**Pull request by manuargue**
This PR introduces support for NXP S32 System Timer Module (STM) driver and enable its usage for `s32z270dc2_r52` boards.

System Timer Module includes a 32-bit count-up timer and four 32-bit compare channels with a separate interrupt source for each channel. The timer is driven by the STM module clock (@ 133 MHz) divided by an 8-bit prescale value.

Updates in NXP HAL: https://github.com/zephyrproject-rtos/hal_nxp/pull/201

### CODE REVIEW

**Review by galak — DISMISSED** [review]
We should really think about the name.  I think most people will take STM to mean ST Micro.  So I think using sys_timer or something like that should be looked at.

At a min, all files / Kconfig symbols, names, etc should get a NXP prefix.

**Comment by manuargue** [comment]
> We should really think about the name. I think most people will take STM to mean ST Micro. So I think using sys_timer or something like that should be looked at.
> 
> At a min, all files / Kconfig symbols, names, etc should get a NXP prefix.

Agree. I've renamed 's32' -> 'nxp_s32' and 'stm' -> 'sys_timer'.

**Comment by manuargue** [comment]
@galak @nordic-krch could you please have another look on this PR? thanks

**Review by nordic-krch — DISMISSED** [review]
lgtm

**Comment by dleach02** [comment]
@galak, can you review to determine if your change request has been satisfied?

**Comment by dleach02** [comment]
HAL PR merged. Please update west.yml

**Comment by manuargue** [comment]
> HAL PR merged. Please update west.yml

manifest updated

**Comment by carlescufi** [comment]
@nordic-krch please take another look

**Review by nordic-krch — DISMISSED** [review]

**Comment by manuargue** [comment]
rebased to fix conflicts

**Review by dleach02 — APPROVED** [review]

File: drivers/counter/counter_nxp_s32_sys_timer.c [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+	DEVICE_DT_DEFINE(SYS_TIMER_NODE(n),						\
+			 nxp_s32_sys_timer_##n##_init,					\
+			 NULL,								\
+			 &nxp_s32_sys_timer_data_##n,					\
+			 &nxp_s32_sys_timer_config_##n,					\
+			 POST_KERNEL,							\
+			 CONFIG_COUNTER_INIT_PRIORITY,					\
+			 &nxp_s32_sys_timer_driver_api);
+
+#if DT_NODE_HAS_STATUS(SYS_TIMER_NODE(0), okay)
```
Comment (danieldegrasse) [inline drivers/counter/counter_nxp_s32_sys_timer.c:253]: Can these initialization macros be replaced with [DT_INST_FOREACH_STATUS_OKAY](https://docs.zephyrproject.org/latest/build/dts/api/api.html#c.DT_INST_FOREACH_STATUS_OKAY)?

File: dts/bindings/timer/nxp,s32-sys-timer.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    description: Module clock frequency in Hz.
+
+  prescaler:
+    type: int
+    required: true
+    description: |
+      Selects the module clock divide value for the prescaler, between 1 and 256.
+      Timer frequency equals to clock-frequency divided by the prescaler value.
+
+  freeze:
```
Comment (danieldegrasse) [inline dts/bindings/timer/nxp,s32-sys-timer.yaml:29]: Nit: I think this property should have `debug` somewhere in the name, to make it clearer what the purpose is without a use needing to read the binding documentation

File: drivers/counter/counter_nxp_s32_sys_timer.c (hunk 1 above)
Comment (manuargue) [reply drivers/counter/counter_nxp_s32_sys_timer.c:253]: The HAL for NXP S32 is index based and in general the interrupt handlers are defined per instance, so the actual hw instance index is still needed. I will address this for all existing drivers so far in another pr, to follow a more general approach. Changes in the baremetal driver will be required in some cases.

File: dts/bindings/timer/nxp,s32-sys-timer.yaml (hunk 2 above)
Comment (manuargue) [reply dts/bindings/timer/nxp,s32-sys-timer.yaml:29]: Right. I used the same name as it appears in the RM of this device.

**Review by nordic-krch — APPROVED** [review]

**Pull request merged**